src/envs/repeated_nextto_painting.py

```diff
@@ -0,0 +1,182 @@
+"""RepeatedNextToPainting domain, which is a merge of our Painting and
+RepeatedNextTo environments.
+
+It is exactly the same as the Painting domain, but requires movement to
+navigate between objects in order to pick or place them. Also, the move
+option can turn on any number of NextTo predicates.
+"""
+
+from typing import Set, Sequence, Dict
+import numpy as np
+from gym.spaces import Box
+from predicators.src.envs.painting import PaintingEnv
+from predicators.src.structs import Predicate, State, \
+    ParameterizedOption, Object, Action, Array
+from predicators.src import utils
+
+
+class RepeatedNextToPaintingEnv(PaintingEnv):
+    """RepeatedNextToPainting domain."""
+
+    def __init__(self) -> None:
+        super().__init__()
+        # Additional Predicates
+        self._NextTo = Predicate("NextTo", [self._robot_type, self._obj_type],
+                                 self._NextTo_holds)
+        self._NextToBox = Predicate("NextToBox",
+                                    [self._robot_type, self._box_type],
+                                    self._NextTo_holds)
+        self._NextToShelf = Predicate("NextToShelf",
+                                      [self._robot_type, self._shelf_type],
+                                      self._NextTo_holds)
+        self._NextToNothing = Predicate("NextToNothing", [self._robot_type],
+                                        self._NextToNothing_holds)
+        # Additional Options
+        self._MoveToObj = ParameterizedOption(
+            "MoveToObj",
+            types=[self._robot_type, self._obj_type],
+            params_space=Box(self.env_lb, self.env_ub, (1, )),
+            _policy=self._Move_policy,
+            _initiable=utils.always_initiable,
+            _terminal=utils.onestep_terminal)
+        self._MoveToBox = ParameterizedOption(
+            "MoveToBox",
+            types=[self._robot_type, self._box_type],
+            params_space=Box(self.env_lb, self.env_ub, (1, )),
+            _policy=self._Move_policy,
+            _initiable=utils.always_initiable,
+            _terminal=utils.onestep_terminal)
+        self._MoveToShelf = ParameterizedOption(
+            "MoveToShelf",
+            types=[self._robot_type, self._shelf_type],
+            params_space=Box(self.env_lb, self.env_ub, (1, )),
+            _policy=self._Move_policy,
+            _initiable=utils.always_initiable,
+            _terminal=utils.onestep_terminal)
+
+    def simulate(self, state: State, action: Action) -> State:
+        assert self.action_space.contains(action.arr)
```

Comment:
I'm wondering if we can make this a little more decoupled from the parent class, and also if we can reduce the duplicated lines here. In both cases, I'm thinking that if (god forbid) we ever want to change the parent class, it would be easy to forget to change the child class correspondingly.

Would the following work? At the beginning of this simulate, compute just the `move_affinity`, and if it's below some threshold, automatically use `self._transition_move`. Otherwise, call `super().simulate(state, action)`.

I see that there are changes to `transition_pick` and `transition_place` too --- but I think we can address those in another way, which I'll comment on separately.